script.js

```diff
@@ -1,4 +1,322 @@
+var mode = "welcome";
+var deck = makeDeck();
+var shuffledDeck = shuffleCards(deck);
+var playerHand = [];
+var dealerHand = [];
+var cardsPerPlayer = 2;
+var playerName = "";
+
+// modes present in game
+// mode: welcome
+function askForName() {
+  return `♦️ ♣️ Welcome to Black Jack! ♠️ ♥️ <br>Please enter your name.`;
+}
+// mode: greet with rules
+function greetWithRules(input) {
+  playerName = input;
+  return `Hi ${playerName}✨! Let's play a round of Black Jack! ♠️ ♥️ ♣️ ♦️ <br> Here are the rules:<br>𐄙 The goal is to beat the dealer's hand without going over 21. <br>𐄚 Each player starts with two cards, one of the dealer's cards is hidden until the end. <br>𐄛 To 'Hit' is to ask for another card. To 'Stand' is to hold your total and end your turn. <br>𐄜 If you go over 21 you bust, and the dealer wins regardless of the dealer's hand. <br>𐄝 If you are dealt 21 from the start (Ace & 10), you got a blackjack.<br>𐄞 Dealer will hit until his/her cards total 17 or higher.<br>💃🏻 Ready to play? Click submit to begin.`;
+}
+// mode: deal cards
+function dealCards() {
+  for (var i = 0; i < cardsPerPlayer; i++) {
+    drawCardInto(playerHand);
+    drawCardInto(dealerHand);
+  }
+  var ace = findAceIn(playerHand);
+  var facesOr10 = findFacesOr10In(playerHand);
+
+  if (ace == true && facesOr10 == true) {
+    mode = "play again?";
+    return `You've got 👯‍♂️ B L A C K  J A C K 👯‍♀️ <br> 🎰 Would you like to play again? (yes/no) 🎰`;
+  } else {
+    return `Your current hand: ${printCardsIn(
+      playerHand
+    )} <br>The dealer's first card is ${dealerHand[0].name} of ${
+      dealerHand[0].suit
+    }<br><br>Would you like to hit or stand? <br>(hit/stand)`;
+  }
+}
+
+// mode: hit
+function hit() {
+  drawCardInto(playerHand);
+  if (sumCardPointsIn(playerHand) > 21) {
+    mode = "play again?";
+    return `You're  🧨 B U S T 💣 <br>Better luck next time 🃏 <br>Would you like to play again? (yes/no)`;
+  } else {
+    return `Your current hand:${printCardsIn(
+      playerHand
+    )} <br>The dealer's first card is ${dealerHand[0].name} of ${
+      dealerHand[0].suit
+    }<br><br>Would you like to hit or stand? <br>(hit/stand)`;
+  }
+}
```

Comment:
It's great that you comment what the code does at the top of the function. One small improvement would be to also comment within the function to make things even clearer. Not necessarily for ALL the lines, but preferably a comment after several chunks of code.

Reply:
This applies to all your functions